src/log.rs

```diff
@@ -32,7 +42,21 @@ pub fn show(mut connection: Connection, verbose: bool) {
                 let topic = publish.topic;
                 let size = publish.payload.len();
                 let payload = Payload::unlimited(publish.payload.into());
-                println!("{time:12} QoS:{qos:11} {topic:50} Payload({size:>3}): {payload}");
+
+                if json {
+                    match serde_json::to_string(&JsonLog {
+                        time,
+                        qos: qos.to_string(),
+                        topic,
+                        size,
+                        payload,
+                    }) {
+                        Ok(out) => println!("{out}"),
+                        Err(err) => eprintln!("Failed to format message: {err}"),
+                    };
+                } else {
+                    println!("{time:12} QoS:{qos:11} {topic:50} Payload({size:>3}): {payload}");
+                };
             }
             Ok(rumqttc::Event::Incoming(packet)) => {
                 if verbose {
```

Comment:
Printing these verbose messages to eprintln might be a good idea. Then only the actual output is on stdout, hints, warnings, verbose and so on on stderr.